From 066304c6b30805a0a1a377534b85e3274b41c23b Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Thu, 27 Aug 2020 16:30:26 +0100
Subject: [PATCH 267/552] media: bcm2835-unicam: Drop WARN on uing direct cache
 alias

Pi 0&1 pass all ARM accesses through the VPU L2 cache, therefore
the dma-ranges property sets the cache alias bits to other
than the direct alias, hence this WARN was firing.

It was overprotective coding, so assume that everything is OK
with the dma-ranges, and remove the WARN.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/platform/bcm2835/bcm2835-unicam.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/media/platform/bcm2835/bcm2835-unicam.c b/drivers/media/platform/bcm2835/bcm2835-unicam.c
index 295db7564614..35ddb97f7f75 100644
--- a/drivers/media/platform/bcm2835/bcm2835-unicam.c
+++ b/drivers/media/platform/bcm2835/bcm2835-unicam.c
@@ -709,13 +709,6 @@ static void unicam_wr_dma_addr(struct unicam_device *dev, dma_addr_t dmaaddr,
 {
 	dma_addr_t endaddr = dmaaddr + buffer_size;
 
-	/*
-	 * dmaaddr and endaddr should be a 32-bit address with the top two bits
-	 * set to 0x3 to signify uncached access through the Videocore memory
-	 * controller.
-	 */
-	WARN_ON((dmaaddr >> 30) != 0x3 || (endaddr >> 30) != 0x3);
-
 	if (pad_id == IMAGE_PAD) {
 		reg_write(dev, UNICAM_IBSA0, dmaaddr);
 		reg_write(dev, UNICAM_IBEA0, endaddr);
-- 
2.33.0

